src/shared/components/Combobox.tsx

```diff
@@ -22,23 +27,25 @@ enum OnChangeEnum {
   createOption = "create-option"
 }
 
-type Props = {
-  options: ComboboxItem[];
+type ComboboxProps = {
+  // but we need to know what data type handles our Combobox otherwise it will fall in case of incorrect data like number[] or object[]
+  // maybe string[] if we know that the combobox should handle this an array of string?
```

Comment:
There are a few techniques that can be used to get the string representation of objects.

The default way to get the label would be to get the tostring result, which works on any object.
Add a property to the combobox called "displayPath" that allows the caller to specify the property to use as a lable.
Add a property called "labelFunction" which is a lambda provided by the caller to convert the object into a text label

Reply:
You are right, but in that case, we need to implement them.
So if we start to use non-string objects we need to add our custom string representation and if we will use some specific methods like toLowerCase (I used to it to get the ID from the label before I changed a possible type to ANY. Now it's removed)

Reply:
In typescript, all value types have a native string representation that can be obtained from object.toString().    Complex type can override .toString to provide a string representation.  The comboxBox needs no special knowledge about the types.  It can simply call .toString on the objects in the itemsSource to get the labels.